pkg/components/bindings/registry.go

```diff
@@ -80,33 +80,61 @@ func (b *bindingsRegistry) RegisterOutputBindings(components ...OutputBinding) {
 }
 
 // Create instantiates an input binding based on `name`.
-func (b *bindingsRegistry) CreateInputBinding(name string) (bindings.InputBinding, error) {
-	if method, ok := b.inputBindings[name]; ok {
+func (b *bindingsRegistry) CreateInputBinding(name, version string) (bindings.InputBinding, error) {
+	if method, ok := b.getInputBinding(name, version); ok {
 		return method(), nil
 	}
-	return nil, errors.Errorf("couldn't find input binding %s", name)
+	return nil, errors.Errorf("couldn't find input binding %s/%s", name, version)
 }
 
 // Create instantiates an output binding based on `name`.
-func (b *bindingsRegistry) CreateOutputBinding(name string) (bindings.OutputBinding, error) {
-	if method, ok := b.outputBindings[name]; ok {
+func (b *bindingsRegistry) CreateOutputBinding(name, version string) (bindings.OutputBinding, error) {
+	if method, ok := b.getOutputBinding(name, version); ok {
 		return method(), nil
 	}
-	return nil, errors.Errorf("couldn't find output binding %s", name)
+	return nil, errors.Errorf("couldn't find output binding %s/%s", name, version)
 }
 
 // HasInputBinding checks if an input binding based on `name` exists in the registry.
-func (b *bindingsRegistry) HasInputBinding(name string) bool {
-	_, ok := b.inputBindings[name]
+func (b *bindingsRegistry) HasInputBinding(name, version string) bool {
+	_, ok := b.getInputBinding(name, version)
 	return ok
 }
 
 // HasOutputBinding checks if an output binding based on `name` exists in the registry.
-func (b *bindingsRegistry) HasOutputBinding(name string) bool {
-	_, ok := b.outputBindings[name]
+func (b *bindingsRegistry) HasOutputBinding(name, version string) bool {
+	_, ok := b.getOutputBinding(name, version)
 	return ok
 }
 
+func (b *bindingsRegistry) getInputBinding(name, version string) (func() bindings.InputBinding, bool) {
+	nameLower := strings.ToLower(name)
+	versionLower := strings.ToLower(version)
+	bindingFn, ok := b.inputBindings[nameLower+"/"+versionLower]
+	if ok {
+		return bindingFn, true
+	}
+	switch versionLower {
+	case "", "v0", "v1":
```

Comment:
nit: the API groups "v0", "v1" etc. are repeated throughout the different registries. maybe its best to consolidate those into well known consts for re-use.

Reply:
See https://github.com/dapr/dapr/pull/2757/commits/8c6ec61670b7355c5ada82c75abbe480d9d04a60